docs/src/DeveloperDocumentation/styleguide.md

```diff
@@ -241,3 +217,213 @@ existing block.
+
+##### Detailed printing with a new line.
+
+```julia
+struct NewRing
+  base_ring
+end
+```
+This implements the detailed printing. Note that the new line is needed for
+this to work. The last print statement must not add a new line.
+
+```julia
+function Base.show(io::IO, ::MIME"text/plain", R::NewRing)
+  println(io, "I am a new ring")
+  println(io, "I print with newlines")
+  print(io, R.base_ring)
+end
+```
+
+This takes care of one line and `:supercompact` printing
+```julia
+function Base.show(io::IO, R::NewRing)
+  if get(io, :supercompact, false)
+    # no nested printing
+    print(io, "supercompact printing of newring ")
+  else
+    # nested printing allowed, preferably supercompact
+    print(io, "one line printing of newring with ")
+    print(IOContext(io, :supercompact => true), "supercompact ", R.base_ring)
+  end
+end
+```
+
+
+##### Detailed printing in a single line.
+
+This version needs to be used in case the detailed
+printing does not contain newlines.
+Then detailed and one line printing agree.
+The if clause takes care of supercompact printing as well.
+```julia
+function Base.show(io::IO, R::NewRing)
+  if get(io, :supercompact, false)
+    # no nested printing
+    print(io, "supercompact printing of newring")
+  else
+    # nested printing allowed, preferably supercompact
+    println(io, "I am a new ring and always print in one line" )
+    print(IOContext(io, :supercompact => true), base_ring(R))
+  end
+end
+```
+
+##### The following is not working as expected and should not be used
+
+```julia
+function Base.show(io::IO, ::MIME"text/plain", R::NewRing)  # do not implement me like this
+  print(io, "I am a new ring with a detailed printing of one line")
+end
+```
+
+Then the following will *not* be used for array/tuple printing.
+It will be used for `print(io, R::NewRing)` though.
+
+```julia
+function Base.show(io::IO, R::NewRing)
+  if get(io, :supercompact, false)
+    print(io, "supercompact printing of newring")
+  else # this is what we call one line
+    print(io, "one line printing of newring with ")
+    print(IOContext(io, :supercompact => true), "supercompact ", R.base_ring)
+  end
+end
+```
+This example illustrates the unexpected behavior.
+```julia
+julia> R = NewRing(1)
+
+julia> R
+I am a new ring with a detailed printing of one line
+
+julia> [R,R]  # one line printing is ignored
+2-element Vector{NewRing}:
+ I am a new ring with a detailed printing of one line
+ I am a new ring with a detailed printing of one line
+
+julia> print(Base.stdout, R)
+one line printing of newring with supercompact QQ
+```
+
+## Latex and unicode printing
+
+### Latex output
+Some types support latex output.
+```
+julia> Qx, x = QQ["x"];
+
+julia> show(stdout, "text/latex", x^2 + 2x + x^10)
+x^{10} + x^{2} + 2 x
+
+julia> show(stdout, "text/latex", Qx[x x^2; 1 1])
+\begin{array}{cc}
+x & x^{2} \\
+1 & 1
+\end{array}
+```
+To support latex output for your type implement
+```
+Base.show(io::IOContext, "text/latex")
```

Comment:
```suggestion
Base.show(io::IOContext, ::MIME"text/latex")
```

Reply:
I trust your word on this. I did not try it out.